core/include/scipp/core/sizes.h

```diff
@@ -27,27 +27,30 @@ class SCIPP_CORE_EXPORT small_stable_map {
   bool operator==(const small_stable_map &other) const noexcept;
   bool operator!=(const small_stable_map &other) const noexcept;
 
-  auto begin() const noexcept { return m_keys.begin(); }
-  auto end() const noexcept { return m_keys.begin() + size(); }
-  auto rbegin() const noexcept { return m_keys.rbegin() - size() + NDIM_MAX; }
-  auto rend() const noexcept { return m_keys.rend(); }
-  typename std::array<Key, Capacity>::const_iterator find(const Key &key) const;
+  [[nodiscard]] auto begin() const noexcept { return m_keys.begin(); }
```

Comment:
Is this seriously the recommended style? `nodiscard` everywhere?

Reply:
Pretty much. I think the guideline is any const function that returns non-void because there is no reason to call such a function without using the return value. (Barring side effects, but such functions should not have any observable side effects.)